File: sdk/php/config/LearningConfig.php
<?php
/******************************************************************************
 *
 *  Copyright 2023 Paphus Solutions Inc.
 *
 *  Licensed under the Eclipse Public License, Version 1.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *      http://www.eclipse.org/legal/epl-v10.html
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 *
 ******************************************************************************/
class LearningConfig extends Config
{
    public ?string $learningMode;
    public ?string $learningRate;
    public ?string $correctionMode;
    public $enableComprehension;
    public $enableEmoting;
    public $enableEmotions;
    public $enableConsciousness;
    public $enableWiktionary;
    public $enableResponseMatch;
    public $learnGrammar;
    public $synthesizeResponse;
    public $fixFormulaCase;
    public $checkExactMatchFirst;
    public int $scriptTimeout;
    public int $responseMatchTimeout;
    public ?string $conversationMatchPercentage;
    public ?string $discussionMatchPercentage;

    public ?string $nlp;
    public ?string $language;
    public ?bool $disableFlag =false;
    public ?bool $reduceQuestions=false;
	public ?bool $trackCase=false;

    public function parseXML($xml): void
    {
        parent::parseXML($xml);
        $xmlData = Utils::loadXML($xml);
        if($xmlData===false) {
            return;
        }
        $this->learningMode = $xmlData->attributes()->learningMode;
        $this->learningRate = $xmlData->attributes()->correctionMode;
        $this->enableComprehension = $xmlData->attributes()->enableComprehension;
        $this->enableEmoting = $xmlData->attributes()->enableEmoting;
        $this->enableEmotions = $xmlData->attributes()->enableEmotions;
        $this->enableConsciousness = $xmlData->attributes()->enableConsciousness;
        $this->enableWiktionary = $xmlData->attributes()->enableWiktionary;
        $this->enableResponseMatch = $xmlData->attributes()->enableResponseMatch;
        $this->learnGrammar = $xmlData->attributes()->learnGrammar;
        $this->synthesizeResponse = $xmlData->attributes()->synthesizeResponse;
        $this->fixFormulaCase = $xmlData->attributes()->fixFormulaCase;
        $this->checkExactMatchFirst = $xmlData->attributes()->checkExactMatchFirst;
        $this->nlp = $xmlData->attributes()->nlp;
        $this->language = $xmlData->attributes()->language;
        $this->disableFlag= $xmlData->attributes()->disableFlag;
        $this->reduceQuestions= $xmlData->attributes()->reduceQuestions;
        $this->trackCase= $xmlData->attributes()->trackCase;


        $value = $xmlData->attributes()->scriptTimeout;
        if (isset($value) && strlen($value) > 0) {
            $this->scriptTimeout = (int) $value;
        }
        $value = $xmlData->attributes()->responseMatchTimeout;
        if (isset($value) && strlen($value) > 0) {
            $this->responseMatchTimeout = (int) $value;
        }
        $this->conversationMatchPercentage = $xmlData->attributes()->conversationMatchPercentage;
        $this->discussionMatchPercentage = $xmlData->attributes()->discussionMatchPercentage;
    }

    public function toXML(): string
    {
        $writer ="";
        $writer .= "<learning";
        $this->writeCredentails($writer);
        if (isset($this->learningMode)) {
            $writer .= " learningMode=\"" . $this->learningMode . "\"";
        }
        if (isset($this->learningMode)) {
            $writer .= " correctionMode=\"" . $this->correctionMode . "\"";
        }
        $writer .= " enableComprehension=\"" . $this->enableComprehension . "\"";
        $writer .= " enableEmoting=\"" . $this->enableEmoting . "\"";
        $writer .= " enableEmotions=\"" . $this->enableEmotions . "\"";
        $writer .= " enableConsciousness=\"" . $this->enableConsciousness . "\"";
        $writer .= " enableWiktionary=\"" . $this->enableWiktionary . "\"";
        $writer .= " enableResponseMatch=\"" . $this->enableResponseMatch . "\"";
        $writer .= " learnGrammar=\"" . $this->learnGrammar . "\"";
        $writer .= " synthesizeResponse=\"" . $this->synthesizeResponse . "\"";
        $writer .= " fixFormulaCase=\"" . $this->fixFormulaCase . "\"";
        $writer .= " checkExactMatchFirst=\"" . $this->checkExactMatchFirst . "\"";
        $writer .= " nlp=\"" . $this->nlp . "\"";
        $writer .= " language=\"" . $this->language . "\"";
        $writer .= " disableFlag=\"" . $this->disableFlag . "\"";
        $writer .= " reduceQuestions=\"" . $this->reduceQuestions . "\"";
        $writer .= " trackCase=\"" . $this->trackCase . "\"";
        if ($this->scriptTimeout != 0) {
            $writer .= " scriptTimeout=\"" . $this->scriptTimeout . "\"";
        }
        if (isset($this->responseMatchTimeout)) {
            $writer .= " responseMatchTimeout=\"" . $this->responseMatchTimeout . "\"";
        }
        if (isset($this->conversationMatchPercentage)) {
            $writer .= " conversationMatchPercentage=\"" . $this->conversationMatchPercentage . "\"";
        }
        if (isset($this->learningRate)) {
            $writer .= " learningRate=\"" . $this->learningRate . "\"";
        }
        $writer .= "/>";
        return $writer;
    }

}
?>

File: sdk/python/config/LearningConfig.py
############################################################################
#
#  Copyright 2023 Paphus Solutions Inc.
#
#  Licensed under the Eclipse Public License, Version 1.0 (the "License");
#  you may not use this file except in compliance with the License.
#  You may obtain a copy of the License at
#
#      http://www.eclipse.org/legal/epl-v10.html
#
#  Unless required by applicable law or agreed to in writing, software
#  distributed under the License is distributed on an "AS IS" BASIS,
#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#  See the License for the specific language governing permissions and
#  limitations under the License.
#
############################################################################
from config.Config import Config
from util.Utils import Utils, Writer
class LearningConfig(Config):
    learningMode: str
    learningRate: str
    correctionMode: str
    enableComprehension: bool
    enableEmoting: bool
    enableEmotions: bool
    enableConsciousness: bool
    enableWiktionary: bool
    enableResponseMatch: bool
    learnGrammar: bool
    synthesizeResponse: bool
    fixFormulaCase: bool
    checkExactMatchFirst: bool
    scriptTimeout: int
    responseMatchTimeout: int
    conversationMatchPercentage: str
    discussionMatchPercentage: str
    nlp: str
    language: str
    disableFlag: bool
    reduceQuestions: bool
    trackCase: bool
    
    def __init__(self):
        super().__init__()
        self.learningMode = None
        self.learningRate = None
        self.correctionMode = None
        self.enableComprehension = False
        self.enableEmoting = False
        self.enableEmotions = False
        self.enableConsciousness = False
        self.enableWiktionary = False
        self.enableResponseMatch = False
        self.learnGrammar = False
        self.synthesizeResponse= False
        self.fixFormulaCase= False
        self.checkExactMatchFirst= False
        self.scriptTimeout = None
        self.responseMatchTimeout = None
        self.conversationMatchPercentage = None
        self.discussionMatchPercentage = None
        self.nlp = None
        self.language = None
        self.disableFlag = False
        self.reduceQuestions = False
        self.trackCase = False
    
    def parseXML(self, xml):
        super().parseXML(xml)
        root = Utils.loadXML(xml)
        if(root == None):
            return 
        self.learningMode = root.attrib.get("learningMode")
        self.learningRate = root.attrib.get("learningRate")
        self.enableComprehension = root.attrib.get("enableComprehension")
        self.enableEmoting = root.attrib.get("enableEmoting")
        self.enableEmotions = root.attrib.get("enableEmotions")
        self.enableConsciousness = root.attrib.get("enableConsciousness")
        self.enableWiktionary = root.attrib.get("enableWiktionary")
        self.enableResponseMatch = root.attrib.get("enableResponseMatch")
        self.learnGrammar = root.attrib.get("learnGrammar")
        self.synthesizeResponse = root.attrib.get("synthesizeResponse")
        self.fixFormulaCase = root.attrib.get("fixFormulaCase")
        self.checkExactMatchFirst = root.attrib.get("checkExactMatchFirst")
        self.nlp = root.attrib.get("nlp")
        self.language = root.attrib.get("language")
        self.disableFlag = root.attrib.get("disableFlag")
        self.reduceQuestions = root.attrib.get("reduceQuestions")
        self.trackCase = root.attrib.get("trackCase")
        
        value = root.attrib.get("scriptTimeout")
        if(value!=None and len(value)>0):
            self.scriptTimeout = int(value)
        value = root.attrib.get("responseMatchTimeout")
        if(value!=None and len(value)>0):
            self.scriptTresponseMatchTimeoutimeout = int(value)
        self.conversationMatchPercentage = root.attrib.get("conversationMatchPercentage")
        self.discussionMatchPercentage = root.attrib.get("discussionMatchPercentage")
    
    def toXML(self):
        writer = Writer("<learning")
        self.writeCredentials(writer)
        if(self.learningMode!=None):
            writer.append(" learningMode=\"" + self.learningMode + "\"")
        if(self.correctionMode!=None):
            writer.append(" correctionMode=\"" + self.correctionMode + "\"")
        writer.append(" enableComprehension=\"" + str(self.enableComprehension) + "\"")
        writer.append(" enableEmoting=\"" + str(self.enableEmoting) + "\"")
        writer.append(" enableEmotions=\"" + str(self.enableEmotions) + "\"")
        writer.append(" enableConsciousness=\"" + str(self.enableConsciousness) + "\"")
        writer.append(" enableWiktionary=\"" + str(self.enableWiktionary) + "\"")
        writer.append(" enableResponseMatch=\"" + str(self.enableResponseMatch) + "\"")
        writer.append(" learnGrammar=\"" + str(self.learnGrammar) + "\"")
        writer.append(" synthesizeResponse=\"" + str(self.synthesizeResponse) + "\"")
        writer.append(" fixFormulaCase=\"" + str(self.fixFormulaCase) + "\"")
        writer.append(" checkExactMatchFirst=\"" + str(self.checkExactMatchFirst) + "\"")
        writer.append(" nlp=\"" + str(self.nlp) + "\"")
        writer.append(" language=\"" + str(self.language) + "\"")
        writer.append(" disableFlag=\"" + str(self.disableFlag) + "\"")
        writer.append(" reduceQuestions=\"" + str(self.reduceQuestions) + "\"")
        writer.append(" trackCase=\"" + str(self.trackCase) + "\"")
        if (self.scriptTimeout!= None and self.scriptTimeout != 0):
            writer.append(" scriptTimeout=\"" + str(self.scriptTimeout) + "\"")
        if (self.responseMatchTimeout != None):
            writer.append(" responseMatchTimeout=\"" + str(self.responseMatchTimeout) + "\"")
        if (self.conversationMatchPercentage != None):
            writer.append(" conversationMatchPercentage=\"" + str(self.conversationMatchPercentage) + "\"")
        if (self.learningRate):
            writer.append(" learningRate=\"" + str(self.learningRate) + "\"")
            
        writer.append("/>")
        return writer